examples/log2har.rs

```diff
@@ -0,0 +1,331 @@
+                        headers: Vec::new(),
+                        content: v1_3::Content {
+                            size: -1,
+                            compression: None,
+                            mime_type: "".to_string(),
+                            text: None,
+                            encoding: None,
+                            comment: None,
+                        },
+                        redirect_url: "".to_string(),
+                        headers_size: -1,
+                        body_size: -1,
+                        comment: None,
+                        headers_compression: None,
+                    },
+                    cache: v1_3::Cache {
+                        before_request: None,
+                        after_request: None,
+                    },
+                    timings: v1_3::Timings {
+                        blocked: None,
+                        dns: None,
+                        connect: None,
+                        send: -1,
+                        wait: -1,
+                        receive: -1,
+                        ssl: None,
+                        comment: None,
+                    },
+                    // TODO - infer from url?
+                    server_ip_address: None,
+                    connection: None,
+                    comment: None,
+                };
+                for element in record.into_inner() {
+                    match element.as_rule() {
+                        Rule::method => entry.request.method = element.as_str().to_string(),
+                        Rule::url => entry.request.url = element.as_str().to_string(),
+                        Rule::request_headers | Rule::curl_headers => {
+                            for header in (ParseHeaders {
+                                iter: element.into_inner(),
+                            }) {
+                                entry.request.headers.push(header)
+                            }
+                        }
+                        // TODO: parse this with more detail
+                        Rule::response_status => {
+                            //println!("{:?}", element)
+                        }
+                        Rule::response_headers => {
+                            for header in (ParseHeaders {
+                                iter: element.into_inner(),
+                            }) {
+                                entry.response.headers.push(header)
+                            }
+                        }
+                        Rule::response_body => {
+                            for pair in element.into_inner() {
+                                match pair.as_rule() {
+                                    Rule::prelude => {}
+                                    Rule::body => {
+                                        entry.response.content.text =
+                                            Some(base64::encode(pair.as_str()));
```

Comment:
It would be great to keep the body as a UTF-8 string (whenever that's the case), because most of our bodies are going to be simple/readable json blobs.  From a quick read of the har spec, I think this is as simple as:

```rust
entry.response.content.text = Some(pair.as_str().to_string());
entry.response.content.encoding = None;  // utf-8 text
```

(I _think_ `pair` is _always_ a valid utf-8 string here, because we start from reading textual log files anyway..)

Reply:
For these examples. Not for the ProtoBuf ones I'm planning on adding next. And not for corner cases like container logs which are ultimately bytestreams, though mojibake may carry the day.

OS^WEditor support for HAR should permit viewing an example interaction with full details even with the encoding; and my experience with nested JSON-in-JSON is that without editor support its actually horribly confusing, because you have a big, one line, massively escaped (all "'s, \n's) blob in the middle of your otherwise structured data. At least as base64 its clearly <this segment> - and its trivial to drag that to base64 -d to extract it, where if its in plain text its escaped and so you have to evaluate that appropriately to get the original back.